refactor(config): Upgrade to winnow 0.5

epage opened the pull request:
Note: this does include a bug fix that a compile error caught.
`one_of(("\n", "\r", "\r\n"))` was actually treating `"\r\n"` as a set
of characters to match and was redundant.  This is now `alt(("\r\n", "\n", "\r"))`

VorpalBlade commented:
Thanks a lot! I'll look into it and do some further testing as soon as I have time.

Meanwhile you might be able to answer a question I had: I added the `.context()` calls, as I understood those would should up in parse errors, but that never seemed to happen (and then I never got time to look into it further). Perhaps I misunderstood that functionality?

epage commented:
> Meanwhile you might be able to answer a question I had: I added the .context() calls, as I understood those would should up in parse errors, but that never seemed to happen (and then I never got time to look into it further). Perhaps I misunderstood that functionality?

You were using `VerboseError` which tried its best to pick a strategy but still had a lot of limitations

> What I would really like (pipe dream sort of stuff perhaps) is if winnow could just make sensible parse errors work automagically (e.g. enable the proper error type, and nicely formatted error messages with context just works). I'm happy to let the library take care of that if it can get it 90% of the way there, it is not something that I'm interested in spending time to get 100% of the way there.

That is my goal with the error type I switched you to that was introduced in winnow 0.5

For example,
```
parse error at line 1, column 9
  |
1 | [package
  |         ^
invalid table
expected ']'
```
- It displays the location of the error and shows you the the line of code in question
- The `Label` is used for the `invalid <label>` line
- The `Expected` variant entries (if present) are joined together for that line

If you've seen a TOML error from cargo in the last while, it followed this pattern.  I took the error type from `toml_edit` and moved it into `winnow`.

VorpalBlade commented:
Code appears to work correctly. I did some performance tests, but the difference (if there is any) is in the noise for my use case (I did some tests with hyperfine).

Parse errors are indeed much better. Thanks!
